### DISCUSSION TITLE
WebDAV不能上传超过几MB的文件

### DISCUSSION THREAD

**Discussion by SucIXR** [AlistGo/alist#7283] [Q&A]
在Ubuntu22.04和openSUSE Tumbleweed部署的Alist都会出现这个问题
测试在任何一种存储方式都会出现
WebDAV挂载在IOS的Documents和KDE的Dolphin上均会出现问题(IOS是没有任何反应,Dolphin反馈"尝试上传XXX"时发生意外错误(0)。"
创建文件夹和上传小文件，以及下载都没问题
在Alist网页上传没问题
这是上传一个文件时的日志:
[GIN] 2024/09/30 - 20:07:47 | 401 |       5.943µs |       127.0.0.1 | PROPFIND  "/dav/共享/未命名文件夹"
[GIN] 2024/09/30 - 20:07:47 | 207 |     120.039µs |       127.0.0.1 | PROPFIND  "/dav/共享/未命名文件夹"
[GIN] 2024/09/30 - 20:07:47 | 401 |       2.377µs |       127.0.0.1 | PROPFIND  "/dav/共享/未命名文件夹"
[GIN] 2024/09/30 - 20:07:47 | 207 |      60.614µs |       127.0.0.1 | PROPFIND  "/dav/共享/未命名文件夹"
[GIN] 2024/09/30 - 20:07:47 | 401 |       2.377µs |       127.0.0.1 | PROPFIND  "/dav/共享/未命名文件夹/大数据.zip"
[33mWARN[0m[2024-09-30 20:07:47] failed get /共享/未命名文件夹/大数据.zip: object not found 
[31mERRO[0m[2024-09-30 20:07:47] PROPFIND /dav/共享/未命名文件夹/大数据.zip object not found
github.com/alist-org/alist/v3/internal/op.Get
	/home/runner/work/alist/alist/internal/op/fs.go:227
github.com/alist-org/alist/v3/internal/fs.get
	/home/runner/work/alist/alist/internal/fs/get.go:38
github.com/alist-org/alist/v3/internal/fs.Get
	/home/runner/work/alist/alist/internal/fs/fs.go:37
github.com/alist-org/alist/v3/server/webdav.(*Handler).handlePropfind
	/home/runner/work/alist/alist/server/webdav/webdav.go:633
github.com/alist-org/alist/v3/server/webdav.(*Handler).ServeHTTP
	/home/runner/work/alist/alist/server/webdav/webdav.go:76
github.com/alist-org/alist/v3/server.ServeWebDAV
	/home/runner/work/alist/alist/server/webdav.go:46
github.com/gin-gonic/gin.(*Context).Next
	/home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/context.go:185
github.com/alist-org/alist/v3/server.WebDAVAuth
	/home/runner/work/alist/alist/server/webdav.go:113
github.com/gin-gonic/gin.(*Context).Next
	/home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/context.go:185
github.com/alist-org/alist/v3/server/middlewares.StoragesLoaded
	/home/runner/work/alist/alist/server/middlewares/check.go:14
github.com/gin-gonic/gin.(*Context).Next
	/home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/context.go:185
github.com/gin-gonic/gin.CustomRecoveryWithWriter.func1
	/home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/recovery.go:102
github.com/gin-gonic/gin.(*Context).Next
	/home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/context.go:185
github.com/gin-gonic/gin.LoggerWithConfig.func1
	/home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/logger.go:249
github.com/gin-gonic/gin.(*Context).Next
	/home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/context.go:185
github.com/gin-gonic/gin.(*Engine).handleHTTPRequest
	/home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/gin.go:633
github.com/gin-gonic/gin.(*Engine).ServeHTTP
	/home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/gin.go:589
net/http.serverHandler.ServeHTTP
	/home/runner/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.4.linux-amd64/src/net/http/server.go:3137
net/http.(*conn).serve
	/home/runner/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.4.linux-amd64/src/net/http/server.go:2039
runtime.goexit
	/home/runner/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.4.linux-amd64/src/runtime/asm_amd64.s:1695 
[GIN] 2024/09/30 - 20:07:47 | 404 |     141.433µs |       127.0.0.1 | PROPFIND  "/dav/共享/未命名文件夹/大数据.zip"
[GIN] 2024/09/30 - 20:07:47 | 401 |       3.566µs |       127.0.0.1 | PUT      "/dav/共享/未命名文件夹/大数据.zip"
[GIN] 2024/09/30 - 20:07:48 | 401 |       2.377µs |       127.0.0.1 | PUT      "/dav/共享/未命名文件夹/大数据.zip"
[GIN] 2024/09/30 - 20:07:48 | 401 |       5.942µs |       127.0.0.1 | PUT      "/dav/共享/未命名文件夹/大数据.zip"
[GIN] 2024/09/30 - 20:07:49 | 401 |       5.943µs |       127.0.0.1 | PUT      "/dav/共享/未命名文件夹/大数据.zip"

**Comment by yyyytawa** [comment]
我的建议是到对于云盘的客户端上传吧,AList本身对于大文件的上传能力不太行....
Webdav基本上就只能拿来访问了,上传性能不太行

**Reply by SucIXR** [reply]
问题是本地存储也有这个问题，去年部署的时候好像没这个问题

**Comment by pongfcnkl** [comment]
反代过？
